src/Core/src/PropertyMapperExtensions.cs

```diff
@@ -145,5 +145,80 @@ void newMethod(IElementHandler handler, IElement view)
 				action?.Invoke(handler!, view);
 			});
 		}
+
+		/// <summary>
+		/// Used internally to provide a fluent API for easily modifying
+		/// mappers at the point of instantiation.
+		/// Specify a method to be run before an existing property mapping.
+		/// </summary>
+		/// <typeparam name="TVirtualView">The cross-platform type.</typeparam>
+		/// <typeparam name="TViewHandler">The handler type.</typeparam>
+		/// <param name="propertyMapper">The property mapper in which to change the mapping.</param>
+		/// <param name="key">The name of the property.</param>
+		/// <param name="method">The method to call before the existing mapping begins.</param>
+		/// <returns>Target Property Mapper</returns>
+		internal static IPropertyMapper<TVirtualView, TViewHandler> Modify<TVirtualView, TViewHandler>(this IPropertyMapper<TVirtualView, TViewHandler> propertyMapper,
+			string key, Action<TViewHandler, TVirtualView, Action<IElementHandler, IElement>?> method)
+			where TVirtualView : IElement where TViewHandler : IElementHandler
+		{
+			propertyMapper.ModifyMapping(key, method);
+			return propertyMapper;
+		}
+
+		/// <summary>
+		/// Used internally to provide a fluent API for easily modifying
+		/// mappers at the point of instantiation.
+		/// Replace a property mapping in place but call the previous mapping if the types do not match.
+		/// </summary>
+		/// <typeparam name="TVirtualView">The cross-platform type.</typeparam>
+		/// <typeparam name="TViewHandler">The handler type.</typeparam>
+		/// <param name="propertyMapper">The property mapper in which to change the mapping.</param>
+		/// <param name="key">The name of the property.</param>
+		/// <param name="method">The modified method to call when the property is updated.</param>
+		/// <returns>Target Property Mapper</returns>
+		internal static IPropertyMapper<TVirtualView, TViewHandler> Replace<TVirtualView, TViewHandler>(this IPropertyMapper<TVirtualView, TViewHandler> propertyMapper,
+			string key, Action<TViewHandler, TVirtualView> method)
+			where TVirtualView : IElement where TViewHandler : IElementHandler
+		{
+			return propertyMapper.Modify(key, (h, v, p) => method.Invoke(h, v));
+		}
```

Comment:
Not sure this method is needed as it just calls the previous which is a replace. If we are going for a symmetric set of APIs, then the non-replacing one needs to apply to: `this IPropertyMapper<IElement, IElementHandler> propertyMapper` (note the `IElement` interface usage) and the generic args are used to the target type like with the public mapper:

```cs
public static void ModifyMapping<TVirtualView, TViewHandler>(this IPropertyMapper<IElement, IElementHandler> propertyMapper,
	string key, Action<TViewHandler, TVirtualView, Action<IElementHandler, IElement>?> method)
	where TVirtualView : IElement where TViewHandler : IElementHandler
{
	var previousMethod = propertyMapper.GetProperty(key);

	void newMethod(IElementHandler handler, IElement view)
	{
		if ((handler is null || handler is TViewHandler) && view is TVirtualView v)
			method((TViewHandler)handler!, v, previousMethod);
		else
			previousMethod?.Invoke(handler!, view);
	}

	propertyMapper.Add(key, newMethod);
}
```

Reply:
Not actually sure the fallback is what we want as the fallback is for checking to see if the view is of a different type. 

For example, if we have a mapper for IView and that does background, then the user/we can "modify" a mapping and add a _more specific_ view to the same mapper. This is used in Controls because the Controls type is a more specific type.

A concrete example is `Label`:

```cs
LabelHandler.Mapper.ModifyMapping<Label, ILabelHandler>(nameof(ILabel.Font), MapFont);
```

This code is adding a new mapper if and only if the type is `Label`. The Core `LabelHandler` only knows about `ILabel` and does some base interaction. In the case of iOS here, when the type is `Label` (aka Controls label type) we want to run some code, and in that code, potentially fall back to the base implementation. The new `MapFont` delegate will NOT run if the type is not `Label` - and this may happen if there is a custom control that just implements `ILabel`.

This is only useful in Core if we have something like a `IFormattedLabel : ILabel` where we want some different code to run if the type is different to the type defined by the mapper type args.